src/tcutility/analysis/report/formatters/xyz.py

```diff
@@ -0,0 +1,122 @@
+import pathlib as pl
+from dataclasses import dataclass
+from typing import List, TypeVar, Union
+
+import scm.plams as plams
+from tcutility import results
+from tcutility.results import Result
+
+
+@dataclass
+class XYZData:
```

Comment:
Do we need this class? I think it will only be used in the formatter class

Reply:
I prefer this way to have a clear(ly typed) datacontainer for the data that is needed for writing an xyz block. An alternative is using a dictionary or tuples with multiple entries, but the python typechecker (Pylance which I think is based on mypy) starts complaining because then it is not clear what's in the dictionary. 

In addition, I struggle with filtering calculations that have not the right task (only single points, geo opts, and transition state searches are currently supported) because the Result object makes a new instance of itself if a key is not present. So, I get either an empty results "{}", or task = None, and it is annoying to deal with these cases., especially if a calculation have failed, or not all data is present (not doing frequencies) because you can get the key in the Results object 1) does not exist but is a dictionary type or 2) does not exist but also not a dictionary type (why???)

That's why I also introduced a lot of try, except blocks and a seperate function to get a property.

Reply:
I agree fully, the Result object is pretty terrible right now. Let's discuss at some point how to improve it! I have some ideas that might be nice and simple to implement.